unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,225 @@
+            self.redis_client_base = redis_client_auto_register
+            self.redis_client_base.is_active = True
+            self.redis_client_base.register(is_blocking=False)
+
+    def test_unregister(self, redis_client_auto_register):
+        """Checks whether the redis.db.publish is called with correct arguments."""
+        self.redis_client = redis_client_auto_register
+        self.redis_client.redis_db.publish = MagicMock()
+        self.redis_client.is_active = True
+        self.redis_client.unregister(is_blocking=False)
+        transaction_id = self.redis_client._blocking_command_responses[
+            "unregister"]["transaction_id"]
+        data = {"name": AREA_ID, "transaction_id": transaction_id}
+        self.redis_client.redis_db.publish.assert_called_once_with(
+            f"{AREA_ID}/unregister_participant",
+            json.dumps(data))
+
+    def test_unregister_is_active_false_throws_exception(self,
+                                                         redis_client_auto_register):
+        """Checks if is active set to false throws exception when user is already unregistered."""
+        with pytest.raises(RedisAPIException,
+                           match="API is already unregistered from the market."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.is_active = False
+            self.redis_client.unregister(is_blocking=True)
+
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_unregister_is_active_true_throws_exception(self,
+                                                        mock_wait_until_timeout_blocking,
+                                                        redis_client_auto_register):
+        """Checks if is active set to true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.is_active = True
+            self.redis_client.unregister(is_blocking=True)
+
+    def test_on_register(self, redis_client_auto_register):
+        """Assigning the correct arguments to blocking command response and
+           checking by calling on_register function with correct message
+           and should not throw an exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "register": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_register(message)
+
```

Comment:
Please also check if is_active is also True ans well as the area_uuid was correctly set
```suggestion
assert redis_client_auto_register.active == True
assert redis_client_auto_register.area_uuid == DEVICE_ID
```

Reply:
Sure, I will also check these things.